app/jenkins-for-jira-ui/src/components/ConnectJenkins/CreateServer/CreateServer.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from 'react';
+import Button, { LoadingButton } from '@atlaskit/button';
+import Form, {
+	FormFooter
+} from '@atlaskit/form';
+import { v4 as uuidv4 } from 'uuid';
+import { useHistory } from 'react-router';
+import {
+	loadingIcon
+} from '../../JenkinsConfigurationForm/JenkinsConfigurationForm.styles';
+import { createJenkinsServer } from '../../../api/createJenkinsServer';
+import { ConfigurationSteps } from '../ConfigurationSteps/ConfigurationSteps';
+import { StyledH1, StyledInstallationContainer, StyledInstallationContent } from '../ConnectJenkins.styles';
+import { generateNewSecret } from '../../JenkinsConfigurationForm/JenkinsConfigurationForm';
+import { isFormValid, setName } from '../../../common/util/jenkinsConnectionsUtils';
+import { ServerConfigurationFormName } from '../../JenkinsConfigurationForm/ServerConfigurationFormElements/ServerConfigurationFormName/ServerConfigurationFormName';
+import { ConnectLogos } from '../ConnectLogos/ConnectLogos';
+
+const CreateServer = () => {
+	const history = useHistory();
+	const [uuid] = useState(uuidv4);
```

Comment:
Do you really need `useState` just to create the uuid since it won't change?  Just `const uuid = uuidv4()` would do wouldn't it?

Reply:
That's a good point. I'll update this.

Reply:
Moved these directly into the API call as they aren't used anywhere else in the component. This way, they are generated when they are needed.